tests/privacy_scenario_3_test.py

```diff
@@ -0,0 +1,257 @@
+Utils.Debug=args.v
+
+testSuccessful=False
+cluster=Cluster(host=TestHelper.LOCAL_HOST, port=TestHelper.DEFAULT_PORT, walletd=True, walletMgr=WalletMgr(True))
+try:
+    TestHelper.printSystemInfo("BEGIN")
+    cluster.killall(allInstances=True, cleanup=True)
+
+    # prod1, prod2, and node1 connected in a mesh
+    # node2 is not started and is only connected to node1
+    topo = { 0 : [1,2], 
+             1 : [0,2], 
+             2 : [0,1], 
+             3 : [2]}
+    Utils.Print("topo: {}".format(json.dumps(topo, indent=4, sort_keys=True)))
+
+    extraNodeosArgs=" --plugin eosio::net_api_plugin "
+
+    if not cluster.launch(pnodes=pnodes, totalNodes=totalNodes, unstartedNodes=unstartedNodes, configSecurityGroup=True, extraNodeosArgs=extraNodeosArgs, topo=topo, printInfo=True):
+        Utils.cmdError("launcher")
+        Utils.errorExit("Failed to stand up eos cluster.")
+    
+    Print("Validating system accounts after bootstrap")
+    cluster.validateAccounts(None)
+
+    cluster.biosNode.kill(signal.SIGTERM)
+
+    prod1 = cluster.getNode(0)
+    prod2 = cluster.getNode(1)
+    securityGroup = cluster.getSecurityGroup()
+
+    # add prod1 and prod2 to turn on security group, in one action *
+    trans = securityGroup.editSecurityGroup(addNodes=[prod1, prod2])
+    # ecord block number (rbn1) transaction was added in
+    trans_block_num = trans["processed"]["block_num"]
+    Print("security group update [[\"node1\", \"node2\"]] published on block {}".format(trans_block_num))
+
+    prod1.waitForIrreversibleBlock(trans_block_num)
+    prod2.waitForIrreversibleBlock(trans_block_num)
+
+    block1 = prod1.getBlock(trans_block_num)
+    block2 = prod2.getBlock(trans_block_num)
+
+    # checking that there is no fork by comparing block ids
+    block1_id = block1["id"]
+    block2_id = block2["id"]
+    # after recorded block number (rbn1) becomes lib, verify production cycle continues without a fork
+    assert block1_id is not None and block1_id == block2_id
+
+    prod1_head    = prod1.getHeadBlockNum()
+    # on this point head should be already ahead of API node #1, but let's add idle wait just to ensure
+    prod1.waitForBlock(prod1_head + 20)
+
+    apiNode1 = cluster.getNode(2)
+    apiNode1_head = apiNode1.getHeadBlockNum()
+    Print("Node1 currently at {}".format(apiNode1_head))
+    Print("producer 0 after security group transaction block became LIB: {}".format(prod1_head))
+    # non-producing nodes do not receive the block that moved lib to or past the recorded block
+    Print("Node1 head = {} Prod1 head = {}".format(apiNode1_head, prod1_head))
+    assert apiNode1_head <= prod1_head
+
+    trans = securityGroup.editSecurityGroup(addNodes=[apiNode1])
+    # record block number (rbn1) transaction was added in
+    trans_block_num = trans["processed"]["block_num"]
+    Print("security group update [[\"node3\"]] published on block ".format(trans_block_num))
+
+    prod1.waitForIrreversibleBlock(trans_block_num)
```

Comment:
Is there a reason you don't use validateSecurityGroup method? If you are not going to use that, you should use waitForTransactionFinalization (it is named something close to that at least) instead of using the transaction block number, since that block number is a guess as to when the transaction is processed.  Since you have 2 producers, when prod1 is the producer, that will be the block number, but if it is prod2's turn to produce, then it may be in the next block or even later.